### ISSUE TITLE
[Bug] [2.2.0] "RangeError: Maximum call stack size exceeded" - Workaround in comments

### ISSUE THREAD

**Issue by Moltkex** [Zettlr/Zettlr#3174]
### Description

Hi, 
I just downloaded **Zettlr-2.2.0-x86_64.AppImage** and when trying to launch it got this:
```
./Zettlr-2.2.0-x86_64.AppImage

(Zettlr:17752): Gtk-WARNING **: 09:11:49.280: Theme parsing error: gtk.css:73:46: The style property GtkScrolledWindow:scrollbars-within-bevel is deprecated and shouldn't be used anymore. It will be removed in a future version
Performing environment check ...
[Application] Pandoc has been bundled with this release. Path: /tmp/.mount_ZettlrsAXj8h/resources/pandoc
Environment check complete.
[17805:0210/091150.988027:ERROR:sandbox_linux.cc(376)] InitializeSandbox() called with multiple threads in process gpu-process.
RangeError: Maximum call stack size exceeded
    at String.split (<anonymous>)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632278)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
    at e (/tmp/.mount_ZettlrsAXj8h/resources/app.asar/.webpack/main/index.js:2:1632458)
```
And Zettlr won't start. The previous version, **Zettlr-2.1.3-x86_64.AppImage** still works with no issues. 

### Reproducing

1. Download Zettlr-2.2.0-x86_64.AppImage
2. Launch the app from terminal **./Zettlr-2.2.0-x86_64.AppImage**

The app should start without issues, but it doesn't 

### Zettlr Version

Stable (most recent version)

### Specify version

Zettlr-2.2.0-x86_64.AppImage

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

EndeavourOS

### Additional Information

Operating System: EndeavourOS
KDE Plasma Version: 5.24.0
KDE Frameworks Version: 5.90.0
Qt Version: 5.15.2
Kernel Version: 5.16.8-arch1-1 (64-bit)
Graphics Platform: X11

**Comment by nathanlesage** [comment]
Ummm … @kyaso? 👀 You had a Linux VM available, right? Because this issue looks *nasty*.

**Comment by kyaso** [comment]
@nathanlesage Yes, I have Ubuntu in a VM, and running Manjaro natively.

I'll have look into it later today.

**Comment by nathanlesage** [comment]
I'm stupid @kyaso Just noticed the first error thing is standard, because Electron has a lot of trouble with GPUs, but that's not a critical error, and none of our concerns. The problem is that RangeError (I've adapted the title so that only the JS error remains)

**Comment by Moltkex** [comment]
> > Just noticed the first error thing is standard, because Electron has a lot of trouble with GPUs, but that's not a critical error, and none of our concerns.
> 
> Ah, I've also just seen it :D
> 
> Btw, I can't reproduce this issue on my Manjaro machine; 2.2.0 AppImage works fine so far. Strange...

Do you use KDE too? I don't know if relevant, just wondering whether it might be ... I have fedora with Gnome too, will try there and report back. Also, might try with a couple of VMs running Debian Testing and openSUSE TW.

**Comment by nathanlesage** [comment]
This is a very similar error to the one we had with the infinite loop in the translation strings yesterday. It's sad that it's an anonymous function though :/

**Comment by TraceyC77** [comment]
I've just encountered this issue today, even though 2.2.0 opened for me yesterday before a system reboot. I don't know if this is relevant, but noting in case, this is the sequence of events since downloading 2.2.0. 

Environment:
Solus Linux, Plasma desktop
Zettlr-2.2.0-x86_64.AppImage

* I had Zettlr 2.1.3 open when I saw an email that 2.2.0 had been released. I downloaded the newer Appimage.
* I closed 2.1.3 and launched 2.2.0, and was working in it yesterday afternon
* When I was done for the evening I shut down the laptop, and booted it up this morning
* When I attempted to launch Zettlr I encountered the error in the description.

I have professional experience as a software tester and have years of Linux experience. I'm happy to provide any information you might need, help troubleshoot and test. Overall I've been very happy with Zetllr :)

**Comment by kyaso** [comment]
@nathanlesage I think it has to do with the debug mode. I disabled debug mode and restarted Zettlr, and then I got the same error. Opened 2.1.3 to re-enable debug mode, and it worked again!

So, hopefully the fix for #3172 should also fix this issue.

To all, here is a temporary mitigation:
1. Download and open Zettlr 2.1.3
2. Enable Debug mode (Preferences -> Advanced -> "Enable debug mode")
3. Close Zettlr
4. Now opening 2.2.0 should work

**Comment by TraceyC77** [comment]
> @nathanlesage, I think it has to do with the debug mode. I disabled debug mode and restarted Zettlr 2.2.0, and then I got the same error. Opened 2.1.3 to re-enable debug mode, then 2.2.0 worked again!
> 
> So, hopefully, the fix for #3172 should also fix this issue.
> 
> To all, here is a temporary (until the next update) mitigation:
> 
>     1. Download and open Zettlr 2.1.3
> 
>     2. Enable Debug mode (Preferences -> Advanced -> "Enable debug mode")
> 
>     3. Close Zettlr
> 
>     4. Now, opening 2.2.0 should work
> 
> 
> **Edit**: Now I can't reliably reproduce it anymore. 2.2.0 opens without problems whether debug mode is enabled or not. But still, enabling debug mode should get Zettlr to boot on all affected installations at least.

I confirmed that this workaround allowed me to open Zettlr 2.2.0. Thanks!

**Comment by nathanlesage** [comment]
@kyaso Possibly! I'll launch a patch some time over the weekend, that should then fix it hopefully!

**Comment by denisoster** [comment]
Arch(from repo and appimage) bug reproduced,

**Comment by nathanlesage** [comment]
@denisoster does the workaround work for you?

**Comment by denisoster** [comment]
@nathanlesage the method described above works

**Comment by sebzeter** [comment]
On Linux, in a bash terminal, one can use 'jq' to enable Debug mode without the need to download and open Zettlr 2.1.3 like this:
```
~> contents="$(jq '.debug = true' ~/.config/Zettlr/config.json)" \
    && echo -E "${contents}" > ~/.config/Zettlr/config.json
```
Now, opening 2.2.0 should work and Debug mode can be unticked (Preferences -> Advanced -> "Enable debug mode")

**Comment by Tettsch** [comment]
As it was not mentioned earlier within this thread, I just wanted to add that I have the same problem with Windows 10. The workaround works for me as well, though. :)

**Comment by Carmina16** [comment]
I have a similar issue with a different function:

```
    at Array.splice (<anonymous>)
    at e (zettlr\resources\app.asar\.webpack\main\index.js:2:1632349)
```

**Comment by nathanlesage** [comment]
Fixed with 2.2.1, released a few minutes ago.

**Comment by Moltkex** [comment]
> Fixed with 2.2.1, released a few minutes ago.

Thanks! I can confirm that the issue's been fixed with 2.2.1; now Zettlr starts and work as expected.  :+1:  :grin:
